### PR TITLE
Caleb/songs list component

### PR DESCRIPTION
**Pull request by calebziamba**
A simple attempt at #11 using ReactNative components.

### CODE REVIEW

File: components/SongList.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  </TouchableOpacity>
+);
+
+const SongList = ({ songs }) => {
+  const LoadSong = (song: Song) =>
+    console.log("load song number ", song.number);
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <FlatList
```
Comment (brandonxia01) [inline components/SongList.tsx:27]: Just for my knowledge, are these standard react-native types to show a list?

**Review by brandonxia01 — APPROVED** [review]
This is epic, Thanks Caleb!

I tested this in my browser and it looks sweet. I did it with a list of 1000 elements too and it remains pretty fast and responsive even in the "low end mobile" setting on chrome.

screenshot for others to see what it looks like:

Screen Shot 2023-03-11 at 4 55 18 PM

File: components/SongList.tsx (hunk 1 above)
Comment (calebziamba) [reply components/SongList.tsx:27]: Yep! Here is the link [for reference](https://reactnative.dev/docs/flatlist )

File: components/SongList.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+const SongItem = ({ song, onPress }) => (
+  <TouchableOpacity onPress={onPress} style={[styles.item]}>
+    <Text style={[styles.title]}>
+      {song.number}. {song.title}
+    </Text>
+    <Text style={[styles.author]}>{song.author}</Text>
+  </TouchableOpacity>
+);
+
+const SongList = ({ songs }) => {
```
[the remark this answers is not in the document]
Comment (calebziamba) [reply components/SongList.tsx:21]: Best I could find was [this relatively old stack overflow question](https://stackoverflow.com/questions/39152877/consider-marking-event-handler-as-passive-to-make-the-page-more-responsive) about it, which suggests it is something of react (or, react-native) issue and is outside of our direct control

**Pull request merged**